src/org/labkey/test/tests/SampleSetTest.java

```diff
@@ -617,6 +633,85 @@ public void deleteLineageParent() throws IOException, CommandException
         dgen.deleteDomain(createDefaultConnection(true));
     }
 
+
+    @Test
+    public void testStringLookupFields() throws IOException, CommandException
+    {
+        String sampleSetName = "10000Samples"; // Testing with 10,000 samples because as per the product the lookup is converted into text field only when the samples exceed 10,000 samples
+        String listName = "MainList";
+
+        goToProjectHome();
+        new PortalHelper(this).addWebPart("Lists");
+
+        log("Creating the sample set of 10000 samples");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSetName);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("label", FieldDefinition.ColumnType.String)));
+        dgen.addDataSupplier("label", () -> dgen.randomString(10))
+                .withGeneratedRows(10000);
+        dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        SaveRowsResponse saveRowsResponse = dgen.insertRows(createDefaultConnection(true), dgen.getRows());
+        log("Successfully  inserted " + saveRowsResponse.getRowsAffected());
+
+        log("Waiting for the sample data to get generated");
+        goToProjectHome();
+        waitAndClickAndWait(Locator.linkWithText(sampleSetName));
+
+        log("Inserting 10,001 row in the sampleset");
+        DataRegionTable table = new DataRegionTable("Material", getDriver());
+        table.clickInsertNewRow();
+
+        setFormElement(Locator.name("quf_Name"), "Sample1");
+        setFormElement(Locator.name("quf_label"), "Sample1");
+        clickButton("Submit");
```

Comment:
Why not just start with 10,001 rows? This just adds unnecessary and costly navigation if we don't do any kind of pre-check before adding the row.

Reply:
The rows(Test data generator)  which get added are random strings and different every time we run the test, so decided to add 10,001 row manually.